base_layer/core/src/chain_storage/lmdb_db/lmdb.rs

```diff
@@ -155,63 +258,136 @@ where K: Serialize {
     }
 }
 
-pub fn lmdb_exists<K>(env: &Environment, db: &Database, key: &K) -> Result<bool, ChainStorageError>
-where K: Serialize {
-    let txn = ReadTransaction::new(env).map_err(|e| ChainStorageError::AccessError(e.to_string()))?;
-    lmdb_exists_txn(&txn, db, key)
-}
-
-pub fn lmdb_len(env: &Environment, db: &Database) -> Result<usize, ChainStorageError> {
-    let txn = ReadTransaction::new(env).map_err(|e| ChainStorageError::AccessError(e.to_string()))?;
+pub fn lmdb_len(txn: &ConstTransaction<'_>, db: &Database) -> Result<usize, ChainStorageError> {
     let stats = txn.db_stat(&db).map_err(|e| {
         error!(target: LOG_TARGET, "Could not read length from lmdb: {:?}", e);
         ChainStorageError::AccessError(e.to_string())
     })?;
     Ok(stats.entries)
 }
 
-pub fn lmdb_iter_next<K, V>(c: &mut Cursor, access: &ConstAccessor) -> Result<(K, V), error::Error>
+pub fn lmdb_fetch_keys_starting_with<V>(
+    key: &str,
+    txn: &ConstTransaction<'_>,
+    db: &Database,
+) -> Result<Vec<V>, ChainStorageError>
 where
-    K: DeserializeOwned,
     V: DeserializeOwned,
 {
-    let (key_bytes, val_bytes) = c.next(access)?;
-    let key = deserialize::<K>(key_bytes)?;
-    let val = deserialize::<V>(val_bytes)?;
-    Ok((key, val))
+    let access = txn.access();
+    let mut cursor = txn.cursor(db).map_err(|e| {
+        error!(target: LOG_TARGET, "Could not get read cursor from lmdb: {:?}", e);
+        ChainStorageError::AccessError(e.to_string())
+    })?;
+
+    info!(target: LOG_TARGET, "Getting rows matching pattern: {}", key);
```

Comment:
Think a few of these logs should be debug! or trace! . I suggest  `trace!` because this is pretty low level and noisy. 
```suggestion
    trace!(target: LOG_TARGET, "Getting rows matching pattern: {}", key);
```